libbeat/scripts/generate_template.py

```diff
@@ -198,6 +198,22 @@ def fill_field_properties(field, defaults, path):
             properties[field.get("name")]["properties"] = prop
 
         dynamic_templates.extend(dynamic)
+    elif field.get("type") == "nested":
+        if len(path) > 0:
+            path = path + "." + field["name"]
+        else:
+            path = field["name"]
+        prop, dynamic = fill_section_properties(field, defaults, path)
+
+        # Only add properties if they have a content
+        if len(prop) is not 0:
+            properties[field.get("name")] = {"type": "nested", "properties": {}}
+            properties[field.get("name")]["properties"] = prop
+
+        dynamic_templates.extend(dynamic)
```

Comment:
@tsg I would expect that we can use path_match for this: https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-templates.html#path-match-unmatch But I didn't test it.